### PR TITLE
Small improvements to Timer

### PR DESCRIPTION
**Pull request by vchuravy**
After much frustration today debugging timer code it boiled down to two things:

1. When `interval` is to small we round to 0 and accidentially create a one-shot timer
2. Errors in the callback are silently dropped.

### CODE REVIEW

File: base/asyncevent.jl [hunk 1]
```diff
@@ -72,7 +75,7 @@ mutable struct Timer
         timeout ≥ 0 || throw(ArgumentError("timer cannot have negative timeout of $timeout seconds"))
         interval ≥ 0 || throw(ArgumentError("timer cannot have negative repeat interval of $interval seconds"))
         timeout = UInt64(round(timeout * 1000)) + 1
-        interval = UInt64(round(interval * 1000))
```
Comment (vtjnash) [inline base/asyncevent.jl:75]: I feel like we should be using `ceil` instead here

File: test/channels.jl [hunk 2]
```diff
@@ -517,6 +517,13 @@ let a = []
     @test a == [1]
 end
 
+# make sure that we don't accidentally create a one-shot timer
+let
+    t = Timer(t->nothing, 10, 0.00001)
```
Comment (vtjnash) [inline test/channels.jl:522]:
```suggestion
    t = Timer(t->nothing, 10, internal=0.00001)
```

File: test/channels.jl [hunk 3]
```diff
@@ -517,6 +517,13 @@ let a = []
...
+    @test ccall(:uv_timer_get_repeat, UInt64, (Ptr{Cvoid},), t) == 1
+    cancel(t)
```
Comment (vtjnash) [inline test/channels.jl:524]:
```suggestion
    close(t)
```
?

**Comment by vchuravy** [comment]
Should I drop 70d5ca8 in favor of #39518?

**Comment by JeffBezanson** [comment]
I think this is fine for now.

File: base/asyncevent.jl [hunk 4]
```diff
@@ -72,7 +75,7 @@ mutable struct Timer
...
+        interval = UInt64(ceil(interval * 1000))
```
Comment (StefanKarpinski) [inline base/asyncevent.jl:78]: Would it make more sense to make sure that `interval` is at least `1` if it is greater than zero on the way in?

**Pull request merged**